lib/config/migration.ts

```diff
@@ -50,7 +33,8 @@ export function migrateConfig(
         optionTypes[option.name] = option.type;
       });
     }
-    const migratedConfig = clone(config) as MigratedRenovateConfig;
+    const newConfig = MigrationsService.run(config);
+    const migratedConfig = clone(newConfig) as MigratedRenovateConfig;
```

Comment:
@rarkins shouldn't the clone be done in migration service?

Reply:
Yes, I think so. @pret-a-porter any reason that won't work?

Reply:
To save order of properties, instead of clone original config we create empty object `const migratedConfig: RenovateConfig = {};` and then fill it step by step go over properties of original config

Reply:
Does it need cloning? And if so, can the clone be done before the `return` statement instead of after?

Reply:
Yeah, we still need cloning. Maybe when all migrations will be refactored will be possible not to use `clone`.

We need cloning, because we use `dequal` to compare original and migrated configs. https://github.com/lukeed/dequal/blob/master/src/index.js under the hood this library by default compares values by `===` (it is not very clear from the docs).

I have tried to do `clone` before statement, but in this case some unit tests failed. I did not investigate deeper, if it is important I will do.

Reply:
Ok, so leave as is for now